### PR TITLE
Pass options to configured TracerProvider

### PR DESCRIPTION
**Pull request by cbandy**
The global TracerProvider missed some arguments to its configured TracerProvider.

I wasn't sure how to describe the placeholder or the test. Suggestions welcome.

I looked for a similar bug in the global MeterProvider and found none, but the implementation was harder for me to follow there. I am certain, however, the correct behavior isn't covered by current tests. I was able to apply this diff and everything still passed:

```diff
diff --git global/internal/meter.go global/internal/meter.go
index afe0f41..f441d7d 100644
--- global/internal/meter.go
+++ global/internal/meter.go
@@ -148,7 +148,7 @@ func (p *meterProvider) Meter(instrumentationName string, opts ...metric.MeterOp
 	defer p.lock.Unlock()
 
 	if p.delegate != nil {
-		return p.delegate.Meter(instrumentationName, opts...)
+		return p.delegate.Meter(instrumentationName)
 	}
 
 	key := meterKey{
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -34,13 +34,14 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
+- Move the `go.opentelemetry.io/otel/api/metric/registry` package into `go.opentelemetry.io/otel/metric/registry` as a part of #1303. (#1316)
 - Move the `Number` type (together with related functions) from `go.opentelemetry.io/otel/api/metric` package into `go.opentelemetry.io/otel/metric/number` as a part of #1303. (#1316)
 - The function signature of the Span `AddEvent` method in `go.opentelemetry.io/otel` is updated to no longer take an unused context and instead take a required name and a variable number of `EventOption`s. (#1254)
 - The function signature of the Span `RecordError` method in `go.opentelemetry.io/otel` is updated to no longer take an unused context and instead take a required error value and a variable number of `EventOption`s. (#1254)
 - Move the `go.opentelemetry.io/otel/api/global` package to `go.opentelemetry.io/otel/global`. (#1262)
 - Rename correlation context header from `"otcorrelations"` to `"baggage"` to match the OpenTelemetry specification. (#1267)
 - Fix `Code.UnmarshalJSON` to work with valid json only. (#1276)
+- Fix the global `TracerProvider` to pass options to its configured provider. (#1329)
```
Comment (MrAlias) [inline CHANGELOG.md:44]: can we move this to the `Fixed` section?

**Review by MrAlias — APPROVED** [review]
:+1:

File: CHANGELOG.md (hunk 1 above)
Comment (cbandy) [reply CHANGELOG.md:44]: Absolutely.

**Review by Aneurysm9 — APPROVED** [review]

**Comment by MrAlias** [comment]
@cbandy sorry, it looks like upstream package reshuffling has caused conflicts here. You should be able to use `otel` instead of `global` and `otel.GetTracerProvider()` instead of `global.TracerProvider()`.

**Comment by cbandy** [comment]
No problem. I can rebase tonight.

**Comment by cbandy** [comment]
Rebased.

**Review by XSAM — APPROVED** [review]

**Review by dashpole — APPROVED** [review]

**Pull request merged**